Assets/MRTK/SDK/Experimental/Dock/Dockable.cs

```diff
@@ -0,0 +1,307 @@
+                Assert.AreEqual(dockedPosition.DockedObject, this, "When a dockable is docked, its dockedPosition reference the dockable.");
+
+                var lerpTime = dockingState == DockingState.Docked ? moveLerpTimeWhenDocked : moveLerpTime;
+
+                if (!isDragging)
+                {
+                    // Don't override dragging
+                    transform.position = Solver.SmoothTo(transform.position, dockedPosition.transform.position, Time.deltaTime, lerpTime);
+                    transform.rotation = Solver.SmoothTo(transform.rotation, dockedPosition.transform.rotation, Time.deltaTime, lerpTime);
+                }
+
+                transform.localScale = Solver.SmoothTo(transform.localScale, dockedPositionScale, Time.deltaTime, lerpTime);
+
+                if (VectorExtensions.CloseEnough(dockedPosition.transform.position, transform.position, distanceTolerance) &&
+                    QuaternionExtensions.AlignedEnough(dockedPosition.transform.rotation, transform.rotation, angleTolerance) &&
+                    AboutTheSameSize(dockedPositionScale.x, transform.localScale.x))
+                {
+                    // Finished docking
+                    dockingState = DockingState.Docked;
+
+                    // Snap to position
+                    transform.position = dockedPosition.transform.position;
+                    transform.rotation = dockedPosition.transform.rotation;
+                    transform.localScale = dockedPositionScale;
+                }
+            }
+            else if (dockedPosition == null && dockingState == DockingState.Undocking)
+            {
+                transform.localScale = Solver.SmoothTo(transform.localScale, originalScale, Time.deltaTime, moveLerpTime);
+
+                if (AboutTheSameSize(originalScale.x, transform.localScale.x))
+                {
+                    // Finished undocking
+                    dockingState = DockingState.Undocked;
+
+                    // Snap to size
+                    transform.localScale = originalScale;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Docks this object in a given <see cref="DockPosition"/>.
+        /// </summary>
+        /// <param name="position">The <see cref="DockPosition"/> where we'd like to dock this object.</param>
+        public void Dock(DockPosition position)
+        {
+            if (!CanDock)
+            {
+                Debug.LogError($"Trying to dock an object that was not undocked. State = {dockingState}");
+                return;
+            }
+
+            Debug.Log($"Docking object {gameObject.name} on position {position.gameObject.name}");
+
+            dockedPosition = position;
+            dockedPosition.DockedObject = this;
+            float scaleToFit = gameObject.GetComponent<Collider>().bounds.GetScaleToFitInside(dockedPosition.GetComponent<Collider>().bounds);
```

Comment:
Might make sense to ensure a Collider component exists in an initialization method. The MRTK has an [EnsureComponent](https://microsoft.github.io/MixedRealityToolkit-Unity/api/Microsoft.MixedReality.Toolkit.ComponentExtensions.html#Microsoft_MixedReality_Toolkit_ComponentExtensions_EnsureComponent_GameObject_Type_) extension to help with this too.